pkg/function/configs.go

```diff
@@ -12,3 +10,20 @@ const (
 	WinStartTime          = "x-numaflow-win-start-time"
 	WinEndTime            = "x-numaflow-win-end-time"
 )
+
+var Protocol string
```

Comment:
How about just defining constants in this file like following, and then in `client.New()`, check the env variable `MAP_MULTIPROC`, and build the different client with these constants?  Using a function like `setupConn()` to change the global variables seems to be weird.

```golang
const (
  TCP = "tcp"
  UDS = "unix"
  UDS_ADDR = "/var/run/numaflow/function.sock"
  TCP_ADDR = ":55551"
)
```

For golang server, it should only use `UDS` and `UDS_ADDR` to start.

Reply:
Updated using this approach